forte/model.py

```diff
@@ -0,0 +1,109 @@
+from abc import ABC, abstractmethod
+from forte.molecule import Molecule
+from forte.basis import Basis
+from forte.forte import StateInfo
+from forte.forte import Symmetry
+
+
+class Model(ABC):
```

Comment:
It'd be worth saying what a model is. I take it this is a model of a system, with a Hamiltonian?

Reply:
I added a tentative description.